solar_system_wave_3.rb

```diff
@@ -0,0 +1,113 @@
+class SolarSystem
+  attr_reader :planets
+  # All planet names should be stored into an instance variable
+  def initialize(planets) # ARRAY OF OBJECTS
+    @planets = planets
+    return
+  end
+  # Have the user create a new planet in the Solar System!
+  def create_planet
+    puts "Name?"
+    user_planet_name = gets.chomp
+    puts "Size?"
+    user_planet_size = gets.chomp
+    puts "Color?"
+    user_planet_color = gets.chomp
+    puts "Type?"
+    user_planet_type = gets.chomp
+    puts "How many moons?"
+    user_planet_moons = gets.chomp
+    # Create a new object if the user wants to create a new planet
+    return Planet.new(user_planet_name, user_planet_size, user_planet_color, user_planet_type, user_planet_moons)
+  end
+  # Add a new planet to the solar system!
+  def add_planet(new_planet)
+    @planets << new_planet
+  end
+  # Print single planet attributes
+  def print_planet_att(planet_choice)
+    # Iterate over each index in array and call the return_att Planet method
+    attributes = ""
+    @planets.each do |planet|
+      if planet.name == planet_choice
+        attributes = planet.return_att
+      # attributes += planet.return_att + "\n"
+      end
+    end
+    return attributes # Return all planet attributes as a long string
+  end
+  # Display all planets in your solar system to give the user choices
+  def display_planet_options
+    planet_names = ""
+    @planets.each do |planet|
+      planet_names += planet.name + "\n"
+    end
+    return planet_names # Return all planet names as a string
+  end
+  # Will print all planets!
+  def print_all_planet_facts
+    @planets.each do |planet|
+      puts "Name:\t#{planet.name}\nSize:\t#{planet.size}\nColour:\t#{planet.colour}\nType:\t#{planet.type}\nNumber of Moons: #{planet.moons}"
```

Comment:
I believe we spoke about this in class but I'd like to get it in writing as well: above you do a great job of having the details of formatting a planet in the `Planet` class, and of returning a string instead of `puts`ing to the console, but here you do not follow that pattern. In my opinion, that makes this code less clean.

Also, hopefully the way we addressed this on the projector in class wasn't too harsh - I ended up nitpicking rather more than I intended. I really appreciate the amount of guts it takes to show something off like that!